__tests__/app.test.js

```diff
@@ -195,21 +196,69 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 
-  test("400: responds with an error message if searched with wrong url. endpoint /api/articles/:article_id/comments", () => {
+  test("400: responds with an error message if searched with wrong url. URL: /api/articles/:article_id/comments", () => {
     return request(app)
       .get("/api/articles/xyz/comment")
       .expect(400)
       .then(({ body }) => {
         expect(body.msg).toEqual("Bad request. Please check what you're requesting and try again.");
       });
-    });
+  });
+
+  test("404: responds with JSON object of all /api/articles/:article_id/comments endpoints", () => {
+    return request(app)
+      .get("/api/articles/1500/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Not Found");
+      });
+  });
+});
+
+//07
+describe("POST /api/articles/:article_id/comments", () => {
+  test("200: responds with a JSON object of the comments of a given article id Which has been posted.", () => {
+    const newComments = {username: 'lurker', body: 'Comments...'};
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComments)
+      .expect(201)
+      .then(({ body }) => {
+        const {comments} = body;
 
-    test("404: responds with JSON object of all /api/articles/:article_id/comments endpoints", () => {
+          expect(comments[0]).toHaveProperty("comment_id", expect.any(Number));
+          expect(comments[0]).toHaveProperty("votes", expect.any(Number));
+          expect(comments[0]).toHaveProperty("created_at", expect.any(String));
+          expect(comments[0]).toHaveProperty("author", expect.any(String));
+          expect(comments[0]).toHaveProperty("body", expect.any(String));
+          expect(comments[0]).toHaveProperty("article_id", expect.any(Number));
+
+          expect(comments[0]["votes"]).toEqual(1);
+          expect(comments[0]["author"]).toEqual('lurker');
+          expect(comments[0]["body"]).toEqual('Comments...');
+      });
+  });
+
+  test("400: responds with an error message if there's a malfunction request, for example missing NOT NULL value/s.", () => {
+  const newComments = {username: 1, body:null}; //Creating violation for the 'NOT NULL' column. Check errors.js for details.
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComments)
+      .expect(400)
+      .then(({ body }) => {
+          expect(body.msg).toMatch("Malfunction: Class 23 — Integrity Constraint Violation:")
+      });
+  });
+
+  test("400: responds with an error message if there's an INCORRECT DATA TYPE in REQUEST BODY.", () => {
+  const newComments = {username: 1, body:'Comments...', votes: 'x'}; //Creating data exception error. votes should be an integer.
```

Comment:
I'm not keen on this test, your code should be written so that your endpoint only accepts username & body as keys, anything else should be ignored and not inserted in the database, things could get dangerous if you let clients insert other keys into the database. Maybe refactor it into a 200 "ignores additional keys" test